workspaces/util/src/__tests__/distinctArray-test.ts

```diff
@@ -0,0 +1,75 @@
+import { describe, expect, it } from "@jest/globals";
+import { distinctArray } from "../distinctArray";
+
+describe("distinctArray", () => {
+    it("returns an empty list if given an empty list", () => {
+        expect(distinctArray([])).toStrictEqual([]);
+    });
+
+    it("can handle NaN in an iterable of numbers", () => {
+        expect(distinctArray([156, 89, NaN, 99, NaN, 156])).toStrictEqual([156, 89, NaN, 99]);
+    });
+
+    it("can handle an iterable of booleans", () => {
+        expect(distinctArray([true, true, false, false])).toStrictEqual([true, false]);
+    });
+
+    it("can handle iterables like sets", () => {
+        expect(
+            distinctArray(
+                new Set(["uno", "dos", "tres", "quatro", "cinco", "cinco", "seis"])
+            )
+        ).toStrictEqual(["uno", "dos", "tres", "quatro", "cinco", "seis"]);
+    });
+
+    it("is not guaranteed to receive distinct result for a map with no keyFn", () => {
+        expect(
+            distinctArray(
+                new Map([[1, "uno"], [2, "dos"], [3, "tres"], [4, "tres"], [5, "cinco!"]])
+            )
+        ).toStrictEqual([[1, "uno"], [2, "dos"], [3, "tres"], [4, "tres"], [5, "cinco!"]]);
+    });
+
+    it("is guaranteed to receive distinct result for a map with a keyFn", () => {
+        expect(
+            distinctArray(
+                new Map([[1, "uno"], [2, "dos"], [3, "tres"], [4, "tres"], [5, "cinco!"]]),
+                (val) => val[1]
+            )
+        ).toStrictEqual([[1, "uno"], [2, "dos"], [4, "tres"], [5, "cinco!"]]);
+    });
+
+    it("can handle iterables like objects", () => {
+        expect(
+            distinctArray(
+                {
+                    *[Symbol.iterator]() {
+                        yield "uno";
+                        yield "dos";
+                        yield "tres";
+                        yield "tres";
+                        yield "cinco!";
+                    }
+                }
```

Comment:
This level of nesting is getting a bit difficult to read. I think we can define the object or generator as its own variable and then use the variable here.